### PR TITLE
Feature/loc group and category bulk

### PR DESCRIPTION
**Pull request by adamkorynta**

### CODE REVIEW

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java [hunk 1]
```diff
@@ -98,7 +102,7 @@ Request createRequest() throws IOException {
     public HttpRequestResponse execute() throws IOException {
         Request request = createRequest();
         ResponseBody body = null;
-        try {
+        try (Timer.Context timer = createTimer().start()) {
```
Comment (rma-psmorris) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:105]: move client creation outside of timer?

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java [hunk 2]
```diff
@@ -132,4 +136,17 @@ public HttpRequestResponse execute() throws IOException {
...
         }
     }
+
+    private Timer createTimer() {
+        if (!CwmsHttpClientMetrics.isMetricsEnabled()) {
+            return new NoOpTimer();
+        }
+        Metrics metrics = CwmsHttpClientMetrics.createMetrics(Objects.toString(httpUrl.resolve(endpoint)));
+        Timer timer = metrics.createTimer();
+        Properties metricsProperties = new Properties();
```
Comment (rma-psmorris) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:146]: bug in metrics api that needs to be resolved, https://jira.hecdev.net/browse/CWMS-734

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LocationGroupEndpointInput.java [hunk 3]
```diff
@@ -42,7 +41,15 @@
     private String categoryId;
 
     public LocationGroupEndpointInput(String groupId) {
-        this.groupId = Objects.requireNonNull(groupId, "Cannot access the location group endpoint without a location group id");
+        this.groupId = groupId;
+    }
+
+    public LocationGroupEndpointInput() {
+        this.groupId = null;
```
[the remark this answers is not in the document]
Comment (rma-psmorris) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LocationGroupEndpointInput.java:50]: you said not supported

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java [hunk 4]
```diff
@@ -59,18 +64,95 @@ void testRetrieveLocationGroup() throws IOException {
         assertEquals("CWMS Mobile Location Listings", locationGroup.getLocationCategory().getId());
 
         assertEquals("FakeLake", locationGroup.getAssignedLocations().get(0).getLocationId());
```
Comment (rma-psmorris) [inline cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java:66]: Doesnt have include assigned set?

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java [hunk 5]
```diff
@@ -59,18 +64,95 @@ void testRetrieveLocationGroup() throws IOException {
...
+        LocationGroup locationGroup = locationGroups.get(1);
+        assertEquals("SWT", locationGroup.getOfficeId());
+        assertEquals("CWMS Standard Naming", locationGroup.getId());
+        assertEquals("Alias group for location IDs that conform to CWMS Standard Naming Convention", locationGroup.getDescription());
+        LocationCategory locationCategory = locationGroup.getLocationCategory();
+        assertEquals("Agency Aliases", locationCategory.getId());
+        assertEquals("CWMS", locationCategory.getOfficeId());
+        assertEquals("Location aliases for other agencies", locationCategory.getDescription());
+
+        List<AssignedLocation> assignedLocations = locationGroup.getAssignedLocations();
```
Comment (rma-psmorris) [inline cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java:155]: not an empty collection>?

**Review by rma-psmorris — CHANGES_REQUESTED** [review]
further review

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java (hunk 5 above)
Comment (adamkorynta) [reply cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java:155]: initializing collections to an empty collection is a feature in the next version of the swagger generator, so for now it's null.

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java (hunk 4 above)
Comment (adamkorynta) [reply cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationGroupController.java:66]: fixed

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:105]: moved

**Review by rma-psmorris — APPROVED** [review]
approved

**Pull request merged**